### PR TITLE
[TVMScript] IRBuilder methods for `Block`

### PR DESCRIPTION
**Pull request by cyx-6**
This PR introduces remaining IRBuilder methods for `Block`.

Co-authored-by: yongwww <yongcale@gmail.com>

### CODE REVIEW

File: python/tvm/script/ir_builder/tir/frame.py [hunk 1]
```diff
@@ -38,6 +38,11 @@ class BlockFrame(TIRFrame):
     ...
```
Comment (junrushao) [inline python/tvm/script/ir_builder/tir/frame.py:38]: please add mypy checks for our IRBuilder package:

In file `tests/scripts/task_mypy.sh`, let's add:

```python
echo "Checking MyPy Type defs in the tvmscript IRBuilder package."
mypy  --check-untyped-defs python/tvm/script/ir_builder
```

Comment (cyx-6) [reply python/tvm/script/ir_builder/tir/frame.py:38]: added

**Comment by junrushao** [comment]
to fix mypy complaints, you may use:

```python
     def __enter__(self) -> "IRBuilderFrame":
-        _ffi_api.IRBuilderFrameEnter(self)  # pylint: disable=no-member # type: ignore
+        _ffi_api.IRBuilderFrameEnter(self)  # type: ignore  # pylint: disable=no-member
         return self

     def __exit__(self, ptype, value, trace) -> None:  # pylint: disable=unused-argument
-        _ffi_api.IRBuilderFrameExit(self)  # pylint: disable=no-member # type: ignore
+        _ffi_api.IRBuilderFrameExit(self)  # type: ignore # pylint: disable=no-member
```

**Review by junrushao — APPROVED** [review]
LGTM!

**Pull request merged**